backend/chatbot/chat_router.py

```
@@ -25,209 +32,143 @@
         )
-
     except HTTPException:
         raise
     except Exception as e:
-        logger.error(f"Chat processing failed: {str(e)}", exc_info=True)
-        raise HTTPException(
-            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
-            detail="Chat processing failed"
-        )
+        logger.error(f"Chat processing failed: {e}", exc_info=True)
+        raise HTTPException(status_code=500, detail="Chat processing failed")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)